cypress/integration/core_opensearch_dashboards/opensearch_dashboards/apps/query_enhancements/s3_dataset.spec.js

```diff
@@ -141,6 +147,40 @@ const s3DatasetTestSuite = () => {
           cy.getElementByTestId('docTable').should('be.visible');
           cy.getElementByTestId('docTable').find('tr').should('have.length', 11);
         });
+
+        it('aborts and cancels previous query when new query is started', function () {
+          cy.getElementByTestId(`datasetSelectorButton`).click();
+          cy.getElementByTestId(`datasetSelectorAdvancedButton`).click();
+
+          cy.get(`[title="S3 Connections"]`).click();
+          cy.get(`[title="BasicS3Connection"]`).click();
+          cy.get(`[title="mys3"]`).click();
+          cy.get(`[title="default"]`).click();
+          cy.get(`[title="http_logs"]`).click();
+          cy.getElementByTestId('datasetSelectorNext').click();
+          cy.get(`[class="euiModalHeader__title"]`).should('contain', 'Step 2: Configure data');
+
+          cy.getElementByTestId('advancedSelectorLanguageSelect').select('OpenSearch SQL');
+          cy.getElementByTestId('advancedSelectorConfirmButton').click();
+
+          // Need to wait a bit for initial query to start
+          cy.wait(3000);
```

Comment:
how come? not blocker but something we should address is avoiding abritraty wait times like a env variable that we can control globally. step 2 would be using other features that help us wait without waiting and assuming everything will happen within 3 secs

like i said tho we shouldn't address this in this pr